python/lsst/meas/extensions/ngmix/LMSimpleShape.py

```diff
@@ -0,0 +1,238 @@
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""
+Definitions and registration of pure-Python plugins with trivial implementations,
+and automatic plugin-from-algorithm calls for those implemented in C++.
+"""
+import ngmix
+from ngmix.fitting import LMSimple, LM_SINGULAR_MATRIX, LM_NEG_COV_EIG, LM_NEG_COV_DIAG,\
+    EIG_NOTFINITE, LM_FUNC_NOTFINITE, LM_DIV_ZERO
+from ngmix.observation import Observation
+
+import lsst.pex.exceptions
+import lsst.afw.detection
+import lsst.afw.geom
+from lsst.shapelet import ShapeletFunction, ShapeletFunctionKey, MultiShapeletFunctionKey, HERMITE
+
+from lsst.meas.base.pluginRegistry import register
+from lsst.meas.base.sfm import SingleFramePluginConfig, SingleFramePlugin
+from lsst.meas.base.baseLib import MeasurementError
+import lsst.meas.base.flagDecorator
+
+__all__ = (
+    "SingleFrameNgmixConfig", "SingleFrameNgmixPlugin"
+)
+
+
+class SingleFrameNgmixConfig(SingleFramePluginConfig):
+
+    psfName = lsst.pex.config.Field(dtype=str, default=None, optional=False,
+                                    doc = "Field name prefix of the PSF approximation plugin to use")
+
+    model = lsst.pex.config.Field(dtype=str, default="gauss", optional=False,
+                                  doc="which LMSimple model to run")
+    maxfev = lsst.pex.config.Field(dtype=int, default=4000, optional=False,
+                                   doc="maximum number of function evaluations")
+    ftol = lsst.pex.config.Field(dtype=float, default=1e-5, optional=False,
+                                 doc="tolerance in sum of squares")
+    xtol = lsst.pex.config.Field(dtype=float, default=1e-5, optional=False,
+                                 doc="tolerance in solution")
+
+
+@register("meas_extensions_ngmix_LMSimpleShape")
+@lsst.meas.base.flagDecorator.addFlagHandler(("flag", "General Failure error"),
+        ("flag_noPsf", "No PSF attached to the exposure."),
+        ("flag_maxFev", "Maximum function evaluations exceeded"),
+        ("flag_singularMatrix", "Singular Matrix"),
+        ("flag_negativeCovEig", "Negative Covariance Eig"),
+        ("flag_negativeCoveDiag", "Negative Covariance Diag"),
+        ("flag_eigNotFinite", "Eig not finite"),
+        ("flag_funcNotFinite", "Func not finite"),
+        ("flag_divideByZero", "Divide by Zero"))
+class SingleFrameNgmixPlugin(SingleFramePlugin):
+    '''
+    Plugin to do shape measurement using ngmix.LMSimple fitter
+    May be run using gauss, dev, or exp model
+    '''
+    ConfigClass = SingleFrameNgmixConfig
+
+    #   Length of a single gaussian representation in ngmix
+    gaussian_pars_len = 6
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return 3.0
+
+    #   This defines the legal LMSimple routines to call
+    #   This metadata will be fetched from Erin when he adds an API for it
+    def getModelInfo(self, model):
+        modelInfo = {"gauss": {"nPars": 6, "nGauss": 1},
```

Comment:
I can see why you would want modelInfo to be a local variable, not poluting the class scope, and that may be best, but it seems the kind of thing that will be created and destroied again and again each function call. Does it make sense to have this a class variable?